mappings/net/minecraft/data/client/model/BlockStateModelGenerator.mapping

```diff
@@ -1,12 +1,422 @@
+		ARG 1 doubleBlock
+		ARG 2 tintType
+	METHOD method_25622 registerSingleton (Lnet/minecraft/class_2248;Lnet/minecraft/class_4946$class_4947;)V
+		ARG 1 block
+		ARG 2 modelFactory
+	METHOD method_25623 registerParentedItemModel (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;)V
+		ARG 1 block
+		ARG 2 parentModelId
+	METHOD method_25626 createFenceGateBlockState (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4917;
+		ARG 0 fenceGateBlock
+		ARG 1 openModelId
+		ARG 2 closedModelId
+		ARG 3 openWallModelId
+		ARG 4 closedWallModelId
+	METHOD method_25630 createEastDefaultHorizontalRotationStates ()Lnet/minecraft/class_4926;
+	METHOD method_25631 registerRotatable (Lnet/minecraft/class_2248;)V
+		ARG 1 block
+	METHOD method_25632 registerBuiltin (Lnet/minecraft/class_2248;Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4911;
+		ARG 1 block
+		ARG 2 particleBlock
+	METHOD method_25633 registerDecorative (Lnet/minecraft/class_2248;Lnet/minecraft/class_4946$class_4947;)Lnet/minecraft/class_4910$class_4912;
+		ARG 1 block
+		ARG 2 modelFactory
+	METHOD method_25634 createBlockStateWithRandomHorizontalRotations (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4925;
+		ARG 0 block
+		ARG 1 modelId
+	METHOD method_25636 createWallBlockState (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4917;
+		ARG 0 wallBlock
+		ARG 1 postModelId
+		ARG 2 lowSideModelId
+		ARG 3 tallSideModelId
+	METHOD method_25640 createNorthDefaultRotationStates ()Lnet/minecraft/class_4926;
+	METHOD method_25641 registerSimpleCubeAll (Lnet/minecraft/class_2248;)V
+		ARG 1 block
+	METHOD method_25642 registerCarpet (Lnet/minecraft/class_2248;Lnet/minecraft/class_2248;)V
+		ARG 1 wool
+		ARG 2 carpet
+	METHOD method_25643 registerUpDefaultFacing (Lnet/minecraft/class_2248;Lnet/minecraft/class_4946$class_4947;)V
+		ARG 1 block
+		ARG 2 modelFactory
+	METHOD method_25644 createSingletonBlockState (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4925;
+		ARG 0 block
+		ARG 1 modelId
+	METHOD method_25645 createBlockStateWithTwoModelAndRandomInversion (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4925;
+		ARG 0 block
+		ARG 1 firstModelId
+		ARG 2 secondModelId
+	METHOD method_25646 createStairsBlockState (Lnet/minecraft/class_2248;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;Lnet/minecraft/class_2960;)Lnet/minecraft/class_4917;
+		ARG 0 stairsBlock
+		ARG 1 innerModelId
+		ARG 2 regularModelId
+		ARG 3 outerModelId
+	METHOD method_25649 createAxisRotatedVariantMap ()Lnet/minecraft/class_4926;
+	METHOD method_25650 registerCubeAllDecorative (Lnet/minecraft/class_2248;)Lnet/minecraft/class_4910$class_4912;
```

Comment:
"cube all decorative"? What does "all" mean here? Should it just be `registerDecorativeCube`?

Reply:
it registers a block with cube_all model and returns a decorative derivative which you can use to register other things like fences, buttons, etc.

Reply:
Done.